### PR TITLE
Executive performance test card implementation

### PR DESCRIPTION
**Pull request by nameisaravind**
This PR is created to add performance test card for portfolio. It includes performance collector and its respective UI changes.

### CODE REVIEW

File: exec-ui/src/app/modules/metrics/modules/metrics/performance-test/services/performance-test.service.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import {MetricDetail} from '../../../shared/domain-models/metric-detail';
```
Comment (Sbrenthughes) [inline exec-ui/src/app/modules/metrics/modules/metrics/performance-test/services/performance-test.service.ts]: I do this too but we need to start defining imports consistently. With angular specific imports at the top and project specific imports below that. Can you make sure we are doing this on all the UI files?

File: exec-analysis/src/main/java/com/capitalone/dashboard/exec/collector/PerformanceCollector.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Arrays;
+import java.util.Date;
+
+@Component
+public class PerformanceCollector extends DefaultMetricCollector {
+
+    private static final String COLLN_PERFORMANCE = "performance";
```
[the remark this answers is not in the document]
Comment (nameisaravind) [reply exec-analysis/src/main/java/com/capitalone/dashboard/exec/collector/PerformanceCollector.java]: constant removed

File: exec-analysis/src/main/java/com/capitalone/dashboard/exec/collector/PerformanceCollector.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    private static final String STR_TIMEWINDOW = "timeWindow";
+    private static final String STR_AVG_RESPONSE_TIME = "averageResponseTime";
+    private static final String STR_CALLSPER_MINUTE = "callsperMinute";
+    private static final String STR_ERROR_RATE = "actualErrorRate";
+    private static final String STR_AVGRESPONSE = "Response Time";
+    private static final String STR_TPS = "TPS";
```
[the remark this answers is not in the document]
Comment (nameisaravind) [reply exec-analysis/src/main/java/com/capitalone/dashboard/exec/collector/PerformanceCollector.java]: TPS - Transaction Per Second. Added comments and it also mentioned in the perf card description.

File: exec-ui/src/app/modules/metrics/modules/metrics/performance-test/services/performance-test.service.ts (hunk 1 above)
Comment (nameisaravind) [reply exec-ui/src/app/modules/metrics/modules/metrics/performance-test/services/performance-test.service.ts]: Import statements reordered in all files.

File: exec-ui/src/app/modules/metrics/modules/metrics/performance-test/strategies/performance-test-preview-strategy.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        const validSet = new Set(['Error Rate']);
+        const sums = model.counts
+            .filter(count => validSet.has(count.label['type']))
+            .reduce((runningSums, count) => {
+                return runningSums.set(count.label['type'], count.value);
+            }, new Map());
+
+        let result = [];
+
+       /* if (sums.get('Transaction Per Second')) {
```
Comment (Sbrenthughes) [inline exec-ui/src/app/modules/metrics/modules/metrics/performance-test/strategies/performance-test-preview-strategy.ts]: Hey what is the reasoning for these being commented out? If we don't need them lets remove them

Comment (nameisaravind) [reply exec-ui/src/app/modules/metrics/modules/metrics/performance-test/strategies/performance-test-preview-strategy.ts]: removed commented lines

**Pull request merged**